### ISSUE TITLE
customized loss function

### ISSUE THREAD

**Issue by rojinsafavi** [ncullen93/torchsample#116]
How can I use a customized loss function as the input to the trainer.compile ?

**Comment by kpoloni** [comment]
I have implemented my loss function by adding a new function in the file metrics.py. 
Example:
```
class MSE(Metric):
    def __init__(self):
        self.mse = 0
        self._name = 'mse'
    def reset(self):
        self.mse = 0
    def __call__(self, y_pred, y_true):
        self.mse = nn.MSELoss()(y_pred, y_true).cpu().detach().numpy()
        return self.mse
```

Then, I edited the function _validate_metric_input (in the modules folder called _utils.py). In this case, I have only added the condition in the function _validate_metric_input:

```
def _validate_metric_input(metric):
    if isinstance(metric, str):
        if metric.upper() == 'CATEGORICAL_ACCURACY' or metric.upper() == 'ACCURACY':
            return CategoricalAccuracy()
        elif metric.upper() == 'BINARY_ACCURACY':
            return BinaryAccuracy()
        # NEW INSERTED CODE
        elif metric.upper() == 'MSE':
            return MSE()
        # END NEW INSERTED CODE
        else:
            raise ValueError('Invalid metric string input - must match pytorch function.')
    elif isinstance(metric, Metric):
        return metric
    else:
        raise ValueError('Invalid metric input')
```


I hope that helps
